pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,207 @@
+#include "lists.h"
+
+int tSizes[8] = {
+            1,
+            sizeof(int8_t), 
+            sizeof(int16_t), 
+            sizeof(int32_t), 
+            sizeof(int64_t), 
+            sizeof(float), 
+            sizeof(double), 
+            sizeof(void *)
+};
+
+t_list   *allocate_list(size_t size, t_type type, void *elemnts) // va_arg could alow us to take in multiple list of elements
+{
+    t_list *list;
+
+    if (!(list = (t_list*)malloc(sizeof(t_list))))
+        return NULL;
+    list->size = size * tSizes[type];
+    list->type = type;
+    list->capacity = DEFAULT_CAP;
+    while (list->capacity <= list->size)
+        list->capacity *= 2;
+    if (list->size == 0)
+        list->elements = NULL;
```

Comment:
@azouiten Do you think `list->elements` should be `NULL` in this case? It make sense if it's allocated with the `DEFAULT_CAP`.

Reply:
yes! that just an oversight.